src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import round from 'lodash/round';
+
+import {t, tn} from 'app/locale';
+import space from 'app/styles/space';
+import UserAvatar from 'app/components/avatar/userAvatar';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import AsyncComponent from 'app/components/asyncComponent';
+import {percent} from 'app/utils';
+import {userDisplayName} from 'app/utils/formatters';
+import {Commit, User} from 'app/types';
+
+import {SectionHeading, Wrapper} from './styles';
+
+type Props = {
+  projectId: string;
+  orgId: string;
+  version: string;
+  commitCount: number;
+} & AsyncComponent['props'];
+
+type State = {
+  commits: Commit[];
+} & AsyncComponent['state'];
+
+class CommitAuthorBreakdown extends AsyncComponent<Props, State> {
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+    };
+  }
+
+  getEndpoints(): [string, string][] {
+    const {projectId, orgId, version} = this.props;
+
+    const commitsEndpoint = `/projects/${orgId}/${encodeURIComponent(
+      projectId
+    )}/releases/${encodeURIComponent(version)}/commits/`;
+
+    return [['commits', commitsEndpoint]];
+  }
+
+  getDisplayPercent(authorCommitCount: number): string {
+    const {commitCount} = this.props;
+
+    const calculatedPercent = round(percent(authorCommitCount, commitCount), 0);
+
+    return `${calculatedPercent < 1 ? '<1' : calculatedPercent}%`;
+  }
+
+  renderBody() {
+    // group commits by author
+    const groupedAuthorCommits = this.state.commits?.reduce(
+      (authorCommitsAccumulator, commit) => {
+        const email = commit.author?.email ?? 'undefined';
```

Comment:
Are you sure about that? is it possible for the authorCommitsAccumulator to have `undefined`?

Reply:
It's a key to group all commits that don't have known authors together. I change the string key into 'unknown' to avoid any confusion.